.github/workflows/validate-pattern-specs.yml

```
@@ -0,0 +1,104 @@
+---
+name: Validate Pattern Spec Version
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+on:
+  pull_request:
+    branches:
+      - main
+      - stable-*
+    tags:
+      - "*"
+jobs:
+  version_check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0 # Fetch full history to allow diffing against previous commits
+      - name: Checkout main branch
+        run: |
+          git fetch origin main:main
+
+      - name: Get changed files
+        id: changed-files
+        run: |
+          # Determine the range of commits to analyze
+          if [ "${{ github.event_name }}" == "pull_request" ]; then
+            BASE_COMMIT=${{ github.event.pull_request.base.sha }}
+            HEAD_COMMIT=${{ github.event.pull_request.head.sha }}
+            CHANGED_FILES=$(git diff --name-only $BASE_COMMIT..$HEAD_COMMIT)
+          else
+            CHANGED_FILES=$(git diff --name-only HEAD^..HEAD)
+          fi
+
+          echo "Changed files: $CHANGED_FILES"
+
+          # Check for pattern specification files (excluding dev versions)
+          SPEC_FILES=$(echo "$CHANGED_FILES" | grep -E '^specifications/pattern-specification/.*\.md$' | grep -v 'dev\.md$' || true)
+          echo "spec_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$SPEC_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+          # Check for pattern schema files (excluding dev versions)
+          SCHEMA_FILES=$(echo "$CHANGED_FILES" | grep -E '^specifications/pattern-schema/.*\.json$' | grep -v 'dev\.json$' || true)
+          echo "schema_files<<EOF" >> $GITHUB_OUTPUT
+          echo "$SCHEMA_FILES" >> $GITHUB_OUTPUT
+          echo "EOF" >> $GITHUB_OUTPUT
+
+          # Check for new pattern specification files that don't exist in main
+          NEW_SPEC_FILES="" # https://community.unix.com/t/if-condition-to-check-one-file-newer-than-the-other-first-file-name-uncertain/243995/7
+          if [ -n "$SPEC_FILES" ]; then
+            echo "$SPEC_FILES" | while read -r file; do
+              if [ -n "$file" ]; then
+                if ! git show main:$file >/dev/null 2>&1' then
+                  echo "$file"
+                fi
+              fi
+            done > /tmp/new_spec_files.txt
+            NEW_SPEC_FILES=$(cat /tmp/new_spec_files.txt 2>/dev/null || true)
```

```suggestion
            NEW_SPEC_FILES=$(cat /tmp/new_spec_files.txt 2>/dev/null || true)
          fi
```
(closing the `if` statement from [line 56](https://github.com/ansible/pattern-service/pull/41/files#diff-34b8f687accfb7bf87b3185c5c1ad28ae03be5aa3b08cc27c18cd3af50f10131R56))